tagsync/src/main/java/org/apache/ranger/tagsync/process/TagSyncMetricsProducer.java

```diff
@@ -18,80 +18,74 @@
-		try {
-			TagSyncConfig config = TagSyncConfig.getInstance();
-			long sleepTimeBetweenCycleInMillis = config.getTagSyncMetricsFrequency();
-			String logFileNameWithPath = config.getTagSyncMetricsFileName();
-			LOG.info("Tagsync metrics frequency :  " + sleepTimeBetweenCycleInMillis +" and metrics file : "+logFileNameWithPath );
-			if (null != logFileNameWithPath) {
-				while (!shutdownFlag) {
-					try {
-						if (LOG.isDebugEnabled()) {
-							LOG.debug("Sleeping Tagsync metrics for [" + sleepTimeBetweenCycleInMillis
-									+ "] milliSeconds");
-						}
-						Thread.sleep(sleepTimeBetweenCycleInMillis);
-					} catch (InterruptedException e) {
-						LOG.error("Failed to wait for [" + sleepTimeBetweenCycleInMillis
-								+ "] milliseconds before attempting to tagsync metrics information", e);
-					}
-					try {
-						writeJVMMetrics(logFileNameWithPath, config);
-					} catch (Throwable t) {
-						LOG.error("Failed to write tagsync metrics into file. Error details: ", t);
-					}
-				}
-			} else {
-				LOG.info("No file directory found for tagsync metrics log ");
-			}
-		} catch (Throwable t) {
-			LOG.error("Failed to start Tagsync metrics. Error details: ", t);
-		} finally {
-			LOG.info("Shutting down the TagSync metrics thread");
-		}
+public class TagSyncMetricsProducer implements Runnable {
+    private static final Logger LOG = LoggerFactory.getLogger(TagSyncMetricsProducer.class);
 
-	}
+    private boolean shutdownFlag;
 
-	private void writeJVMMetrics(String logFileNameWithPath, TagSyncConfig config) throws Throwable {
-		try {
-			File userMetricFile = null;
-			userMetricFile = new File(logFileNameWithPath);
-			if (!userMetricFile.exists()) {
-				userMetricFile.createNewFile();
-			}
-			RangerMetricsUtil rangerMetricsUtil = new RangerMetricsUtil();
-			if (config.getBoolean(TagSyncConfig.TAGSYNC_SERVER_HA_ENABLED_PARAM, false)) {
-				if(config.isTagSyncServiceActive()){
-					rangerMetricsUtil.setIsRoleActive(1);
-				}else{
-					rangerMetricsUtil.setIsRoleActive(0);
-				}
-			}
-			rangerMetricsUtil.writeMetricsToFile(userMetricFile);
+    public static void main(String[] args) {
+        TagSyncMetricsProducer tagSyncMetrics = new TagSyncMetricsProducer();
+        tagSyncMetrics.run();
+        // try { tagSyncMetrics.writeJVMMetrics(); } catch (Throwable e) { }
+    }
 
-		} catch (Throwable t) {
-			LOG.error("TagSyncMetricsProducer.writeJVMMetrics() failed to write metrics into file. Error details: ", t);
-			throw t;
-		}
-	}
+    @Override
+    public void run() {
+        try {
+            TagSyncConfig config                        = TagSyncConfig.getInstance();
+            long          sleepTimeBetweenCycleInMillis = config.getTagSyncMetricsFrequency();
+            String        logFileNameWithPath           = config.getTagSyncMetricsFileName();
+            LOG.info("Tagsync metrics frequency :  {} and metrics file : {}", sleepTimeBetweenCycleInMillis, logFileNameWithPath);
```

Comment:
please keep a line space between initializations and log messages.

Reply:
@kumaab 
I have addressed it here in the latest commit.
But I think we should add it as a checkstyle rule if we want to sustain this. 
Looking at the code, some places look like spaces before/after logs may be a good choice for readability but other places not so much (but if we have this defined as a rule then it makes sense since it will be uniform)

Reply:
@fateh288  - we will add checkstyle rule if/when they are available. Until then, it is okay to require extra effort in keeping the code base consistent.

Reply:
@mneethiraj Agree